src/main/java/ca/ciccc/BankAccount.java

```diff
@@ -75,72 +76,110 @@ public void withdraw(double amount) {
      *
      * @param fee
      * @return true if there's enough balance, otherwise false
      */
     public boolean transactionFee(double fee) {
-        // TODO 2: Your code goes here.
+        double totalfee = 0;
+
+        for (int i = 1; i < transactionCount; i++) {
+            totalfee = totalfee + (i * fee);
+
+            balance = balance - totalfee;
+            if (balance < 0) {
+                return false;
+            }
+
+        }
+        return true;
 
-        return false;
     }
 
+
     /**
      * The transfer method moves money from `this` bank account to another account. The method
      * accepts two parameters: the amount of money to transfer and the other ca.ciccc.BankAccount to accept
      * the money.
      * There is a $5.00 fee for transferring money, so this much must be deducted from the current
      * account's balance before any transfer.
-     *
+     * <p>
      * If `this` account object does not have enough money to make full transfer, then transfer
      * whatever money is left after the $5.00 fee is deducted. If this account has under $5.00 or amount is 0 or less,
      * no transfer should occur and neither account's state should be modified.
-     *
+     * <p>
      * If any amount of money is transferred, return {@code true}. Otherwise {@code false}.
      *
      * @param amount
      * @param other
      * @return true if transferred any amount of money, otherwise false.
      */
-    public boolean transfer(double amount, BankAccount other) {
-        // TODO 3: Your code goes here.
+    public String transfer(double amount, BankAccount other) {
+        double fee = 5.00;
+        double total = 0;
+        double BankAccount = this.getBalance();
+        for (int i = 1; i < transactionCount; i++) {
+            total = amount + (i * fee);
+            if (BankAccount - total >= 0) {
+                this.withdraw(fee);
+                this.withdraw(amount);
+                other.withdraw(amount);
+                other.deposit(amount);
+                return String.valueOf(true);
 
-        return false;
-    }
 
-    /**
-     * Your {@code toString()} method should return a string that contains the account's name
-     * and balance separated by a comma and space. For example, if an account object has name
-     * "Derrick" and a balance of 20.55, this method should return "Derrick, $20.55"
-     * There are some special cases you should handle. If the balance is negative, put the -sign
-     * before the dollar sign. Also, always display the cents as a two-digit number.
-     * For example, if the same object had a balance of -17.5, your method should return
-     * "Derrick, -$17.50"
-     *
-     * @return string representation of ca.ciccc.BankAccount
-     */
-    @Override
-    public String toString() {
-        // TODO 1: Your code goes here.
+            } else if (BankAccount - total <= 0) {
+
+                return String.valueOf(false);
```

Comment:
[FYI] You can just return `false`.

```suggestion
                return false;
```